rdtools/test/analysis_chains_test.py

```diff
@@ -131,6 +131,39 @@ def test_sensor_analysis_ad_hoc_filter(sensor_parameters):
         rd_analysis.sensor_analysis(analyses=['yoy_degradation'])
 
 
+def test_filter_components(sensor_parameters):
+    poa = sensor_parameters['poa_global']
+    poa_filter = (poa > 200) & (poa < 1200)
+    rd_analysis = TrendAnalysis(**sensor_parameters, power_dc_rated=1.0)
+    rd_analysis.sensor_analysis(analyses=['yoy_degradation'])
+    assert (poa_filter == rd_analysis.sensor_filter_components['poa_filter']).all()
+
+
+def test_filter_components_no_filters(sensor_parameters):
+    rd_analysis = TrendAnalysis(**sensor_parameters, power_dc_rated=1.0)
+    rd_analysis.filter_params = {}  # disable all filters
+    rd_analysis.sensor_analysis(analyses=['yoy_degradation'])
+    expected = pd.Series(True, index=rd_analysis.pv_energy.index)
+    pd.testing.assert_series_equal(rd_analysis.sensor_filter, expected)
+    assert rd_analysis.sensor_filter_components.empty
+
+
+def test_filter_ad_hoc_warnings(sensor_parameters):
```

Comment:
For both of the cases in this test, I suggest checking that the final `rd_analysis.sensor_filter` is as expected. At least by counting Trues and Falses. If not an explicit comparison.

Reply:
I've added checks that the ad_hoc_filter is processed as described in the warning messages but didn't test the final `sensor_filter` value as I think other tests verify that the `ad_hoc_filter` is incorporated correctly into the final filter.  What do you think?

Reply:
That works, thanks!